client/acme.go

```diff
@@ -407,6 +409,25 @@ func (d *dns01P2PForgeSolver) Wait(ctx context.Context, challenge acme.Challenge
 func (d *dns01P2PForgeSolver) Present(ctx context.Context, challenge acme.Challenge) error {
 	h := d.hostFn()
 	addrs := h.Addrs()
+
+	if !d.allowPrivateForgeAddresses {
```

Comment:
@aschmahmann doing this in `Present` is too late, it is called in the middle of the ACME dance, and we want to avoid the entire ACME flow if we are not publicly reachable. 

Pushed https://github.com/ipshipyard/p2p-forge/pull/4/commits/5157ed9b6bf68416338384c351b0eace0d80d95e which is tackling problem at a higher level: it is delaying calling `ManageAsync` for respective domain/cert  in the top level `func (m *P2PForgeCertMgr) Start() error`, and the delay occurs only when necessary (no cert cached), and does not occur when you already have a cert.

It seems to do the trick for my Kubo setup (ipv4-only, NAT + port forwarding with uPnP) and also gives us framework for plugging up more nuanced checks if needed (right now it just waits for `network.ReachabilityPublic`).

Reply:
Pushed further refactor in https://github.com/ipshipyard/p2p-forge/pull/4/commits/d638643497d6fa08e2b468c9efc83a232ef104cd which extracts connectivity checks to separate reusable func `withHostConnectivity`.